src/llmq/snapshot.cpp

```diff
@@ -0,0 +1,263 @@
+
+    UniValue obj3c;
+    quorumSnapshotAtHMinus3C.ToJson(obj3c);
+    obj.pushKV("quorumSnapshotAtHMinus3C", obj3c);
+
+    UniValue objdifftip;
+    mnListDiffTip.ToJson(objdifftip);
+    obj.pushKV("mnListDiffTip", objdifftip);
+
+    UniValue objdiffc;
+    mnListDiffAtHMinusC.ToJson(objdiffc);
+    obj.pushKV("mnListDiffAtHMinusC", objdiffc);
+
+    UniValue objdiff2c;
+    mnListDiffAtHMinus2C.ToJson(objdiff2c);
+    obj.pushKV("mnListDiffAtHMinus2C", objdiff2c);
+
+    UniValue objdiff3c;
+    mnListDiffAtHMinus3C.ToJson(objdiff3c);
+    obj.pushKV("mnListDiffAtHMinus3C", objdiff3c);
+}
+
+bool BuildQuorumRotationInfo(const CGetQuorumRotationInfo& request, CQuorumRotationInfo& response, std::string& errorRet)
+{
+    AssertLockHeld(cs_main);
+
+    if (request.baseBlockHashesNb > 4) {
+        errorRet = strprintf("invalid requested baseBlockHashesNb");
+        return false;
+    }
+
+    if (request.baseBlockHashesNb != request.baseBlockHashes.size()) {
+        errorRet = strprintf("missmatch requested baseBlockHashesNb and size(baseBlockHashes)");
+        return false;
+    }
+
+    LOCK(deterministicMNManager->cs);
+
+    //Quorum rotation is enabled only for InstantSend atm.
+    Consensus::LLMQType llmqType = Params().GetConsensus().llmqTypeInstantSend;
+
+    std::vector<const CBlockIndex*> baseBlockIndexes;
+    if (request.baseBlockHashesNb == 0) {
+        const CBlockIndex *blockIndex = ::ChainActive().Genesis();
+        if (!blockIndex) {
+            errorRet = strprintf("genesis block not found");
+            return false;
+        }
+        baseBlockIndexes.push_back(blockIndex);
+    }
+    else {
+        for (const auto& blockHash : request.baseBlockHashes){
+            const CBlockIndex* blockIndex = LookupBlockIndex(blockHash);
+            if (!blockIndex) {
+                errorRet = strprintf("block %s not found", blockHash.ToString());
+                return false;
+            }
+            if (!::ChainActive().Contains(blockIndex)){
+                errorRet = strprintf("block %s is not in the active chain", blockHash.ToString());
+                return false;
+            }
+            baseBlockIndexes.push_back(blockIndex);
+        }
+        std::sort(baseBlockIndexes.begin(), baseBlockIndexes.end(), [](const CBlockIndex* a, const CBlockIndex* b){
+            return a->nHeight < b->nHeight;
+        });
+    }
+
+    const CBlockIndex* tipBlockIndex = ::ChainActive().Tip();
+    if (!tipBlockIndex) {
+        errorRet = strprintf("tip block not found");
+        return false;
+    }
+    //Build MN list Diff always with highest baseblock
+    if (!BuildSimplifiedMNListDiff(baseBlockIndexes.back()->GetBlockHash(), tipBlockIndex->GetBlockHash(), response.mnListDiffTip, errorRet)) {
+        return false;
+    }
+
+    const CBlockIndex* blockIndex = LookupBlockIndex(request.blockRequestHash);
+    if (!blockIndex) {
+        errorRet = strprintf("block not found");
+        return false;
+    }
```

Comment:
Neither `request.blockRequestHash` or `blockIndex` is used after this point.

The returned `qrinfo` is based on the current tip of the blockchain, rather than what was requested in the `getqrinfo` message.

Reply:
Fixed